source/robot_model/src/Model.cpp

```diff
@@ -13,28 +13,33 @@ Model::Model(const std::string& robot_name,
              const std::string& urdf_path,
              const std::optional<std::function<std::string(const std::string&)>>& meshloader_callback
              ):
-    robot_name_(std::make_shared<state_representation::Parameter<std::string>>("robot_name", robot_name)),
-    urdf_path_(std::make_shared<state_representation::Parameter<std::string>>("urdf_path", urdf_path)),
+    robot_name_(robot_name),
+    urdf_path_(urdf_path),
     meshloader_callback_(meshloader_callback),
     load_collision_geometries_(true)
     {
   this->init_model();
 }
 
 Model::Model(const std::string& robot_name, const std::string& urdf_path) :
-    robot_name_(std::make_shared<state_representation::Parameter<std::string>>("robot_name", robot_name)),
-    urdf_path_(std::make_shared<state_representation::Parameter<std::string>>("urdf_path", urdf_path))
+    robot_name_(robot_name),
+    urdf_path_(urdf_path)
     {
   this->init_model();
 }
 
-Model::Model(const Model& model) :
-    robot_name_(model.robot_name_),
-    urdf_path_(model.urdf_path_), 
-    meshloader_callback_(model.meshloader_callback_),
-    load_collision_geometries_(model.load_collision_geometries_)
+Model::Model(const Model& other):
+    robot_name_(other.robot_name_),
+    urdf_path_(other.urdf_path_),
+    robot_model_(other.robot_model_),
+    robot_data_(other.robot_data_),
+    geom_model_(other.geom_model_),
```

Comment:
just to point out, the Pinocchio `GeometryModel` holds a vector of `GeometryObject`s which are stored as shared pointers to `fcl::CollisionGeometry`. This means that the copy refers to the same collision data in memory. In this particular case, a shallow copy is preferable because we only ever mutate the collision geometry when the robot model is constructed, and this way we don't have to duplicate the geometry in memory (which can get quite heavy). However, we must keep this in mind if we ever decide to add functions to modify collision geometry post-construction!

Reply:
I see! Thank you for the note. With that, it is more clear to me when to use deep copy/shallow copy of objects in memory.